app/javascript/components/ansible-credentials-form/index.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { API } from '../../http_api';
+import MiqFormRenderer from '@@ddf';
+import createSchema from './ansible-credentials-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const AnsibleCredentialsForm = ({ recordId }) => {
+  const [{ fields, initialValues, isLoading }, setState] = useState({ fields: [], isLoading: !!recordId });
+  const promise = useMemo(() => API.options('/api/authentications'), []);
+  const submitLabel = !!recordId ? __('Save') : __('Add');
+
+  const loadSchema = (value, appendState = {}) => ({
+    data: {
+      credential_types: { embedded_ansible_credential_types },
+    },
+  }) =>
+    setState(state => ({
+      ...state,
+      ...appendState,
+      fields: embedded_ansible_credential_types[value].attributes,
+    }));
+
+  useEffect(() => {
+    if (recordId) {
+      API.get(`/api/authentications/${recordId}`).then((initialValues) => {
+        promise.then(loadSchema(initialValues.type, { initialValues, isLoading: false }));
+      });
+    }
+  }, []);
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+
+    const request = recordId ? API.patch(`/api/authentications/${recordId}`, values) : API.post('/api/authentications', values);
```

Comment:
I created an [issue](https://github.com/ManageIQ/manageiq-api/issues/919) to track the problem that the API is returning us a non-error `200` response if the request failed. I hope we can get a systematic solution for that, so the `catch` below will just do its magic out of the box.